src/frontend/Sider/Behandling/Inngangsvilkår/Målgruppe/Målgruppe.tsx

```diff
@@ -14,34 +14,34 @@ import { Feilmelding } from '../../../../komponenter/Feil/Feilmelding';
 import { VilkårPanel } from '../../../../komponenter/VilkårPanel/VilkårPanel';
 import { FlexColumn } from '../../../../komponenter/Visningskomponenter/Flex';
 import { paragraflenkerMålgruppe, rundskrivMålgruppe } from '../../lenker';
-import { VilkårperioderGrunnlag } from '../typer/vilkårperiode';
+import { VilkårperioderGrunnlag, YtelseGrunnlagPeriode } from '../typer/vilkårperiode';
 import VilkårperiodeRad from '../Vilkårperioder/VilkårperiodeRad';
 
 const Målgruppe: React.FC<{ grunnlag: VilkårperioderGrunnlag | undefined }> = ({ grunnlag }) => {
     const { settUlagretKomponent, nullstillUlagretKomponent } = useApp();
     const { målgrupper } = useInngangsvilkår();
     const { erStegRedigerbart } = useSteg();
 
-    const [leggerTilNyPeriode, settLeggerTilNyPeriode] = useState<boolean>(false);
-    const [radIRedigeringsmodus, settRadIRedigeringsmodus] = useState<string>();
+    const [radSomRedigeres, settRadSomRedigeres] = useState<string>();
     const [feilmelding, settFeilmelding] = useState<string>();
+    const [periodeFraRegister, settPeriodeFraRegister] = useState<
+        YtelseGrunnlagPeriode | undefined
+    >(undefined);
 
     const fjernRadIRedigeringsmodus = () => {
         settFeilmelding(undefined);
-        settRadIRedigeringsmodus(undefined);
-        settLeggerTilNyPeriode(false);
+        settRadSomRedigeres(undefined);
+        settPeriodeFraRegister(undefined);
         nullstillUlagretKomponent(UlagretKomponent.MÅLGRUPPE);
     };
 
-    const kanSetteNyRadIRedigeringsmodus =
-        radIRedigeringsmodus === undefined && !leggerTilNyPeriode;
+    const kanSetteNyRadIRedigeringsmodus = radSomRedigeres === undefined && erStegRedigerbart;
```

Comment:
Har endret så i stedet for å ha to states for å passe på om noe redigeres, så er det kun 1 og så brukes "nyPeriode" som id hvis det er snakk om ny periode. Usikker på om dette er lurt eller ikke, men forenklet slik at man kun trengte en metode for å "sette noe i redigeringsmodus"

Reply:
Lurt synes jeg